src/handles/handleDiagnostic.ts

```diff
@@ -55,27 +56,133 @@ function getSecurity(
   return security !== undefined ? security : 1
 }
 
+function handleLinterRegex(output: string, config: ILinterConfig): Diagnostic[] {
+  const {
+    formatLines = 1,
+    formatPattern,
+    offsetLine = 0,
+    offsetColumn = 0,
+    securities = {},
+    sourceName
+  } = config
+  let diagnostics: Diagnostic[] = [];
+
+  if (!formatLines || !formatPattern) {
+    throw new Error('missing formatLines or formatPattern')
+  }
+
+  const { line, column, endLine, endColumn, message, security } = formatPattern[1]
+  const endLineLookup = endLine != null ? endLine : line
+  const endColumnLookup = endColumn != null ? endColumn : column
+  const lines = output.split('\n')
+
+  let str: string = lines.shift()
+  while(lines.length > 0 || str !== undefined) {
+    str = [str].concat(lines.slice(0, formatLines - 1)).join('\n')
+    const m = str.match(new RegExp(formatPattern[0]))
+    if (m) {
+      let diagnostic: Diagnostic = {
```

Comment:
Now that there are multiple linter handlers, there's the chance to simplify these results:

- strip out `getSecurity`
- strip out `sumNum`
- strip out `sourceName`
- handle missing `endLine` and `endColumn` automatically

Since all the info required for those is available at the calling point as well. Could look something like this:


```ts
const result: ILinterResult = {
  security: m[security],
  line: m[line],
  column: m[column],
  endLine: endLine != null ? m[endLine] : undefined
  endColumn: endColumn != null ? m[endColumn] : undefined
  message: formatMessage(message, m)
}
```